src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -176,8 +176,9 @@ def _delete_bid(self, payload):
 
     def _delete_bid_impl(self, arguments, response_channel):
         try:
+            self.state.offered_buy_kWh[self.market.time_slot] -= \
+                self.market.bids[arguments['bid']].energy
             self.remove_bid_from_pending(arguments["bid"], self.market.id)
-            self.state.offered_buy_kWh[self.market.time_slot] -= arguments["bid"]["energy"]
```

Comment:
I think the fix needs to be ported to the offer validation too, because I see in the delete_offer_implementation an `arguments["offer"]["energy"]`, which is obviously incorrect.